bot/bot.py

```diff
@@ -1,52 +1,39 @@
 import asyncio
 import contextlib
-from sys import exception
-
+from sys import exc_info
 import aiohttp
 from discord.errors import Forbidden
 from pydis_core import BotBase
 from pydis_core.utils.error_handling import handle_forbidden_from_block
-from sentry_sdk import new_scope, start_transaction
-
+from sentry_sdk import push_scope, start_transaction
 from bot import constants, exts
 from bot.log import get_logger
 
 log = get_logger("bot")
 
-
 class StartupError(Exception):
     """Exception class for startup errors."""
-
     def __init__(self, base: Exception):
-        super().__init__()
+        super().__init__(str(base))
         self.exception = base
 
-
 class Bot(BotBase):
     """A subclass of `pydis_core.BotBase` that implements bot-specific functions."""
 
-    def __init__(self, *args, **kwargs):
-
-        super().__init__(*args, **kwargs)
-
     async def load_extension(self, name: str, *args, **kwargs) -> None:
         """Extend D.py's load_extension function to also record sentry performance stats."""
         with start_transaction(op="cog-load", name=name):
             await super().load_extension(name, *args, **kwargs)
 
     async def ping_services(self) -> None:
         """A helper to make sure all the services the bot relies on are available on startup."""
-        # Connect Site/API
-        attempts = 0
-        while True:
+        for attempt in range(1, constants.URLs.connect_max_retries + 1):
             try:
-                log.info(f"Attempting site connection: {attempts + 1}/{constants.URLs.connect_max_retries}")
+                log.info(f"Attempting site connection: {attempt}/{constants.URLs.connect_max_retries}")
                 await self.api_client.get("healthcheck")
-                break
-
+                return
             except (aiohttp.ClientConnectorError, aiohttp.ServerDisconnectedError):
-                attempts += 1
-                if attempts == constants.URLs.connect_max_retries:
+                if attempt == constants.URLs.connect_max_retries:
                     raise
```

Comment:
We can make use of the `StartupError` exception here and raise it instead of a bare exception.